src/test/java/org/dmfs/jems/iterable/decorators/MappedTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.iterable.decorators;
+
+import org.dmfs.iterables.EmptyIterable;
+import org.dmfs.iterables.elementary.Seq;
+import org.dmfs.jems.function.Function;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import static org.dmfs.jems.hamcrest.matchers.IterableMatcher.iteratesTo;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * Unit test for {@link Mapped}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class MappedTest
+{
+
+    @Test
+    public void test_emptyIterable()
+    {
+        assertThat(new Mapped<>(new Function<Object, Object>()
```

Comment:
Why not pass a `dummy(Function.class)`?

Reply:
True, that's simpler, I've updated it.
(Inlined version caused compilation-like error in IDE when run.)